### PR TITLE
Apexmalhar 2017

### PR DESCRIPTION
**Pull request by prasannapramod**
@tweise @PramodSSImmaneni please review. Originally submitted by pramod. Multiple tests were failing with latest, so fixed tests.

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java [hunk 1]
```diff
@@ -41,29 +26,32 @@
...
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java:35]: I don't see any net change in this file.

File: contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisOutputOperator.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
  * @param <T>
  * @since 2.0.0
  */
-public abstract class AbstractKinesisOutputOperator<V, T> implements Operator
+public abstract class AbstractKinesisOutputOperator<V, T> implements Operator, Operator.CheckpointNotificationListener
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisOutputOperator.java:52]: Why this change?

File: library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java [hunk 3]
```diff
@@ -79,7 +79,7 @@
  */
 @OperatorAnnotation(checkpointableWithinAppWindow = false)
 @Deprecated
-public class FileSplitter implements InputOperator, Operator.CheckpointListener
+public class FileSplitter implements InputOperator, Operator.CheckpointListener, Operator.CheckpointNotificationListener
```
Comment (tweise) [inline library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java:82]: Is this necessary for backward compatibility of just because of japicmp? If the latter, can the change be whitelisted?

**Review by tweise — CHANGES_REQUESTED** [review]
It will be good if more folks familiar with respective operators review these changes, especially for the file output.

Please also see http://apex.apache.org/contributing.html on how to open PRs.

**Comment by tweise** [comment]
Please remove all changes from the PR that are unrelated to the checkpoint listener (formatting, arranging imports etc.). That can be done in a separate PR, in fact it would be great if you could take up fixing the checkstyle in the contrib module: https://issues.apache.org/jira/browse/APEXMALHAR-1909

**Comment by prasannapramod** [comment]
@tweise sorry I pushed wrong branch. I will close this & create a new pull request shortly.
